### DISCUSSION TITLE
Separate Outline View

### DISCUSSION THREAD

**Discussion by BarryDrez** [eclipse-theia/theia#10427] [Q&A] [answered]
Hi, I would like to create my own version of an outline view for the Java editor.  I do not want to contribute to the existing Java Editor Outline view.  I'd like to keep that one as is and have a separate view that has my own contributions.  Is this possible?  Are there any examples?  Thank you.

**Comment by vince-fugnitto** [comment, marked as answer]
@BarryDrez thank you for the discussion. 

If I understand correctly, you do not want to replace the current generic outline-view (right-most widget in the screenshot) but implement your own? It is definitely possible, but there are two choices. You can either implement the feature as a custom theia extension, or you can implement it as a vscode extension (plugin). The latter would likely be easier to start off with, and there might be examples or existing extensions that can help you achieve your use-case. 

In terms of if you want to go with a theia extension, you'll want to create a [custom theia extension](https://theia-ide.org/docs/authoring_extensions/) and likely use the existing [outline-view](https://github.com/eclipse-theia/theia/tree/master/packages/outline-view) as a reference.

<img width="1226" alt="Screen Shot 2021-11-18 at 8 05 30 AM" src="https://user-images.githubusercontent.com/40359487/142421318-69f80e00-8bdf-423e-b22a-eac870288d68.png">

If you have other more specific questions please don't hesitate to ask 👍

**Reply by BarryDrez** [reply]
Hi @vince-fugnitto,
Thanks a lot for your well written reply.  You are correct that I want to implement my own version of an Outline View.  I will try as you suggest by writing a vscode plugin and see how that goes.  I'll let you know how it works out.

**Reply by BarryDrez** [reply]
It worked out well.  Thanks again.
